src/Mono.Android/Java.Interop/JavaConvert.cs

```diff
@@ -56,6 +57,17 @@ static class JavaConvert {
 
 		static Func<IntPtr, JniHandleOwnership, object?>? GetJniHandleConverter (Type? target)
 		{
+			const string justification = "JavaDictionary<,>, JavaList<>, and JavaCollection<> use DynamicallyAccessedMembers for PublicMethods to preserve FromJniHandle().";
+			[UnconditionalSuppressMessage ("Trimming", "IL2055", Justification = justification)]
+			[UnconditionalSuppressMessage ("Trimming", "IL2068", Justification = justification)]
```

Comment:
It looks like the warnings were about public methods:
```
src\Mono.Android\Java.Interop\JavaConvert.cs(68,5): error IL2068: 'MakeGenericType(Type, params Type[])' method return value does not satisfy 'DynamicallyAccessedMemberTypes.PublicMethods' requirements. The parameter 'type' of method 'MakeGenericType(Type, params Type[])' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.
src\Mono.Android\Java.Interop\JavaConvert.cs(68,5): error IL2055: Call to 'System.Type.MakeGenericType(params Type[])' can not be statically analyzed. It's not possible to guarantee the availability of requirements of the generic type.
```
There is some code below that would call:

```csharp
typeof(JavaList<T>).GetMethod ("FromJniHandle", BindingFlags.Static | BindingFlags.Public)
```

I added `[DAM(PublicMethods)]` on `JavaList<T>`, does that seem like it would work for now?

I think we are good on the generic parameters on the three types, I have `[DAM(Ctors)]` for generics on `JavaList<T>`, `JavaCollection<T>`, and `JavaDictionary<K,V>`:

* https://github.com/xamarin/xamarin-android/pull/8778/files#diff-60dc6156fcc89d17dc9ae7ae9fd821c59bfc4d75a6830b36054c130b2d69d964R688
* https://github.com/xamarin/xamarin-android/pull/8778/files#diff-3c160b3d5560b1c37d5d0bb5e97f21ab78e6238d71558552142b56e513797d02R404
* https://github.com/xamarin/xamarin-android/pull/8778/files#diff-60dc6156fcc89d17dc9ae7ae9fd821c59bfc4d75a6830b36054c130b2d69d964R29

Warnings for these appeared from other places.

Reply:
Sorry - that's not how it works.
Putting DAM on a type only works if you have a variable statically declared as that type and then call `.GetType()` on the variable. Otherwise it will NOT apply the DAM to the type. The goal of DAM is to make things verifiable... so if you still need to use suppressions, it means you probably got it wrong.

The first warning should go away if you change the local method to:
```
			static Type MakeGenericType ([DynamicallyAccessedMembers(PublicMethods)] Type type, params Type [] typeArguments) 
```

You should also remove the DAM annotations on the specific types (JavaList and so on).

The second one doesn't seem to be fixable though:
```C#
class TestType
{
    public TestType() {}
}

// Internally this will create:
//     JavaList<TestType> instance, but that requires TestType to keep the public constructor
//  But the trimmer will remove the constructor, since it doesn't see TestType used anywhere
//  where it would need a constructor.
GetJniHandleConverter(typeof(IList<TestType>));
```

There's no good solution as-is... DI had the same problem and didn't really solve it either. It workaround this by checking the DAM annotations on the generic parameters of the interface at runtime, but that would not help here either, since we can't put the annotation on the interface. I don't know what to do with this second one. But I don't understand the higher level of this, so maybe there's something we can do there.